core/src/main/java/hudson/model/ParametersAction.java

```diff
@@ -67,6 +67,16 @@
 @ExportedBean
 public class ParametersAction implements RunAction2, Iterable<ParameterValue>, QueueAction, EnvironmentContributingAction, LabelAssignmentAction {
 
+    /**
+     * Three state variable (null, false, true).
+     *
+     * If explicitly set to true, it will keep all variable, explicitly set to
+     * false it will drop all of them (except if they are marked safe).
+     * If null, and they are not safe, it will log a warning in logs to the user
+     * to let him choose the behavior
+     *
+     * @since TODO
```

Comment:
Merge/cleanup note: Supported setting to "true" from 1.651.2 / 2.3, supported setting "false" from TODO.